core/src/main/scala/kafka/controller/KafkaController.scala

```diff
@@ -1253,7 +1253,7 @@ class KafkaController(val config: KafkaConfig, zkUtils: ZkUtils, val brokerState
         // If delete topic is disabled remove entries under zookeeper path : /admin/delete_topics
         for (topic <- topicsToBeDeleted) {
           info("Removing " + getDeleteTopicPath(topic) + " since delete topic is disabled")
-          zkUtils.zkClient.delete(getDeleteTopicPath(topic))
+          zkUtils.deletePath(getDeleteTopicPath(topic))
```

Comment:
Sorry I wanted to contact you about that part of the code. I think this one is irrelevant because this part of the code where we are handling the exception is a dead. I think we should remove it.
If we check the zkClient method:

public boolean delete(final String path, final int version) {
        try {
            retryUntilConnected(new Callable<Object>() {

                @Override
                public Object call() throws Exception {
                    _connection.delete(path, version);
                    return null;
                }
            });

            return true;
        } catch (ZkNoNodeException e) {
            return false;
        }
    }
The ZkNoNodeException is handled inside the zkClient. I thought maybe an older version of the zkClient does not handled this exception, so I checked back until version 0.1.0 and even that old version of the client handles this exception.